src/functions/fast-login.ts

```diff
@@ -196,6 +199,33 @@ export const getAssertionUserIdVsCfVerifier = (
     TE.map(() => true as const)
   );
 
+const deleteNonce: (
+  lollipopHeaders: LollipopHeaders
+) => RTE.ReaderTaskEither<
+  FnLollipopClientDependency,
+  H.HttpError,
+  true
+> = lollipopHeaders => ({ redisClientTask }) =>
+  pipe(
+    TE.Do,
+    TE.bind("redisClient", () =>
+      pipe(redisClientTask, TE.mapLeft(errorToHttpError))
+    ),
+    TE.chainW(({ redisClient }) =>
+      pipe(
+        getNonceFromSignatureInput(lollipopHeaders["signature-input"]),
+        TE.fromEither,
+        TE.chain(nonce => getDeleteNonce(nonce)(redisClient)),
+        TE.mapLeft(
+          error =>
+            new CustomHttpUnauthorizedError(
+              `Could not delete nonce: [${error.message}]`
+            )
+        )
+      )
+    )
+  );
```

Comment:
Verify the request shape before creating the database connection is more efficient on malformed requests.
```suggestion
  pipe(
    TE.fromEither(
      getNonceFromSignatureInput(lollipopHeaders["signature-input"])
    ),
    TE.mapLeft(
      error =>
        new CustomHttpUnauthorizedError(
          `Invalid or missing nonce in request: [${error.message}]`
        )
    ),
    TE.chain(nonce =>
      pipe(
        redisClientTask,
        TE.mapLeft(errorToHttpError),
        TE.chainW(
          flow(
            invalidate(nonce),
            TE.mapLeft(
              error =>
                new CustomHttpUnauthorizedError(
                  `Could not delete nonce: [${error.message}]`
                )
            )
          )
        )
      )
    )
  );
```

Reply:
done in https://github.com/pagopa/io-functions-fast-login/pull/27/commits/afa278b7e7d61fa946dc35679329250566193fdd